File: clients/python/examples/tables-create-text-index-exists.py
from astrapy import DataAPIClient

# Get an existing table
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
table = database.get_table("**TABLE_NAME**")

# Index a column
table.create_text_index(
    "**INDEX_NAME**",
    column="**TEXT_COLUMN_NAME**",
    if_not_exists=True,
)

File: clients/java/examples/tables-create-text-index-exists.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.tables.Table;
import com.datastax.astra.client.tables.commands.options.CreateTextIndexOptions;
import com.datastax.astra.client.tables.definition.indexes.TableTextIndexDefinition;
import com.datastax.astra.client.tables.definition.rows.Row;

public class Example {

  public static void main(String[] args) {
    // Get an existing table
    Table<Row> table =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getTable("**TABLE_NAME**");

    // Index a column
    CreateTextIndexOptions options = new CreateTextIndexOptions().ifNotExists(true);
    TableTextIndexDefinition definition =
        new TableTextIndexDefinition().column("**TEXT_COLUMN_NAME**");
    table.createTextIndex("**INDEX_NAME**", definition, options);
  }
}
